modules/home-manager/ghostty.nix

```
@@ -0,0 +1,24 @@
+{ catppuccinLib }:
+{ config, lib, ... }:
+
+let
+  inherit (config.catppuccin) sources;
+
+  cfg = config.catppuccin.ghostty;
+  themeName = "catppuccin-${cfg.flavor}";
+in
+{
+  options.catppuccin.ghostty = catppuccinLib.mkCatppuccinOption { name = "ghostty"; };
+
+  config = lib.mkIf cfg.enable {
```

```suggestion
  config = lib.mkIf enable {
```
And the way we usually do that is just by guarding all of our changes behind that condition

---

I want to confirm this one, because I actually had that implemented as you suggest and was told it was unnecessary here: https://github.com/catppuccin/nix/pull/446#discussion_r1901586957

---

At that revision the comment was correct due to the module only touching `programs.ghostty` options

Those are already only applied when `programs.ghostty.enable` is `true`. When we touch options outside of it (in this case `xdg.configFile`) though, we need to implement that same guard internally (else we may end up placing port files on people's systems for programs they're not actually using)

---

okie dokie.